src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPInterfaceProperties.cs

```diff
@@ -0,0 +1,199 @@
+
+        public override IPAddressInformationCollection AnycastAddresses
+        {
+            get
+            {
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public override IPAddressCollection DnsAddresses { get { return _dnsAddresses; } }
+
+        public override GatewayIPAddressInformationCollection GatewayAddresses { get { return _gatewayAddresses; } }
+
+        public override IPAddressCollection DhcpServerAddresses
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public override IPAddressCollection WinsServersAddresses
+        {
+            get
+            {
+                // TODO: Can we return false? Is this Windows-specific?
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public override IPv4InterfaceProperties GetIPv4Properties()
+        {
+            return _ipv4Properties;
+        }
+
+        public override IPv6InterfaceProperties GetIPv6Properties()
+        {
+            return _ipv6Properties;
+        }
+
+        private static string GetDnsSuffix()
+        {
+            string data = File.ReadAllText(LinuxNetworkFiles.EtcResolvConfFile);
+            RowConfigReader rcr = new RowConfigReader(data);
+            string dnsSuffix;
+
+            if (rcr.TryGetNextValue("search", out dnsSuffix))
+            {
+                return dnsSuffix;
+            }
+            else
+            {
+                return string.Empty;
+            }
+        }
+
+        private UnicastIPAddressInformationCollection GetUnicastAddresses()
+        {
+            var collection = new UnicastIPAddressInformationCollection();
+            foreach (IPAddress address in _linuxNetworkInterface.Addresses.Where((addr) => !IsMulticast(addr)))
+            {
+                IPAddress netMask = (address.AddressFamily == AddressFamily.InterNetwork)
+                                    ? _linuxNetworkInterface.GetNetMaskForIPv4Address(address)
+                                    : IPAddress.Any; // Windows compatibility
+                collection.InternalAdd(new LinuxUnicastIPAddressInformation(address, netMask));
+            }
+
+            return collection;
+        }
+
+        private MulticastIPAddressInformationCollection GetMulticastAddresses()
+        {
+            var collection = new MulticastIPAddressInformationCollection();
+            foreach (IPAddress address in _linuxNetworkInterface.Addresses.Where(IsMulticast))
+            {
+                collection.InternalAdd(new LinuxMulticastIPAddressInformation(address));
+            }
+
+            return collection;
+        }
+
+        private static bool IsMulticast(IPAddress address)
+        {
+            if (address.AddressFamily == AddressFamily.InterNetworkV6)
+            {
+                return address.IsIPv6Multicast;
+            }
+            else
```

Comment:
/cc @pgavlin @stephentoub 
I'm not sure. (It's inferred by the [example](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) in coding-style.md. )

I used this style too but I was told to remove them.